content/articles/how-to-create-font-chooser-app/index.md

```diff
@@ -0,0 +1,139 @@
+
+Then we give our font app a name (I used `maji.ico`. You can use your own name if you want to customize the app.). Then we set the size of the application.
+
+```python
+def font_chooser(a):
+user_font.config(
+ family=my_listbox.get(my_listbox.curselection()))
+```
+
+Then we write the `font_chooser` function, which includes defining the font function and configuring the `Listbox` to retrieve the system's current font selection.
+
+```python
+user_font = font.Font(family="Helvetica", size="30")
+
+my_frame = Frame(root, width=490, height=285)
+my_frame.pack(pady=10)
+my_frame.grid_propagate(False)
+my_frame.columnconfigure(0, weight=10)
+```
+
+Then we specify our font by setting our `font_variable` to a Helvetica typeface with a size of 30. Then a frame is created by defining the `my_frame` variable, which specifies the width and height.
+
+The `.grid` function is also used to freeze the frame and prevent the size of the `Listbox` from changing unnecessarily each time a new font is picked from the font chooser app.
+
+```python
+user_text = Text(my_frame, font=user_font)
+user_text.grid(row=1, column=1)
+user_text.grid_rowconfigure(0, weight=1)
+user_text.grid_columnconfigure(0, weight=1)
+
+user_listbox = Listbox(root, selectmode=SINGLE, width=90)
+user_listbox.pack()
+```
+
+By defining the `user_text` variable, we can add the textbox. We do this by putting the defined variable in a frame and utilising our font to make the font size dynamic.
+
+We utilize the `.grid` function to prevent the font app from resizing abnormally, and we set the row and column weights to 1 to ensure that it evenly distributes any additional space in the row and column.
+
+We create a `Listbox` by setting the `my_listbox` variable to a listbox with a single-mode selection, which allows us to choose only one font at a time while still specifying the width size.
+
+```python
+for f in font.families():
+  my_listbox.insert('end', f)
+
+
+my_listbox.bind('<ButtonRelease-1>', font_chooser)
+root.mainloop()
+```
+
+We use a `for` loop to iterate over the `font.families` sequence to add font families to the already formed `Listbox`, and then use the `.bind` function to bind the `Listbox`.
+
+`mainloop()` is the method that we will use to run the font chooser application. It will allow the application to run continuously and indefinitely until the user decides to end the program by closing the window or using the Escape key on the keyboard.
+
+And the font chooser interface will look like this:
+![fontapp image](/engineering-education/how-to-create-font-chooser-application/fontchooser.png)
+
+### Conclusion
+In this tutorial, we reviewed the role of Tkinter in data science and built a font chooser application.
+
+### Reference
+[Python GUI programming with Tkinter](https://www.perlego.com/book/721869/python-gui-programming-with-tkinter-pdf)
+
```

Comment:
Please include the author of this book in this reference. ie. `Python GUI programming with Tkinter - By author name`